lib/auth/in_memory/backend.js

```diff
@@ -3,16 +3,37 @@
 const crypto = require('crypto');
 
 const errors = require('../../errors');
-const accountsKeyedbyAccessKey =
-    require('./vault.json').accountsKeyedbyAccessKey;
-const accountsKeyedbyCanID =
-    require('./vault.json').accountsKeyedbyCanID;
-const accountsKeyedbyEmail =
-    require('./vault.json').accountsKeyedbyEmail;
 const calculateSigningKey = require('./vaultUtilities').calculateSigningKey;
 const hashSignature = require('./vaultUtilities').hashSignature;
+const Indexer = require('./indexer');
+
+function _buildArn(generalResource, specificResource) {
+    return `arn:aws:s3:::${generalResource}/${specificResource}`;
```

Comment:
I made an underlying class that also takes an indexer / formatter as constructor parameters since it's possible we'll want to extend the backend / indexer for Mystique. Not sure, just in case.